Sql-math-function-floor (#2250)
Codecademy/docs: content/sql/concepts/math-functions/terms/floor/floor.md

---
Title: 'FLOOR()'
Description: 'Rounds down a number to the nearest whole integer.'
Subjects:
  - 'Data Science'
  - 'Computer Science'
Tags:
  - 'SQL'
  - 'MySQL'
  - 'PostgreSQL'
  - 'SQLite'
  - 'Functions'
CatalogContent:
  - 'learn-sql'
  - 'paths/analyze-data-with-sql'
---

> The SQL math function `FLOOR()` returns the largest integer that is less than or equal to a specified numeric value. It effectively rounds down a number to the nearest whole integer.

Reviewer: ```suggestion
The SQL math function **`FLOOR()`** returns the largest integer that is less than or equal to a specified numeric value. It effectively rounds down a number to the nearest whole integer.
```

## Syntax

```sql
SELECT FLOOR(number)
FROM table_name;

> ```

Reviewer 2: Like to see a sentence explanation of the syntax describing what `number` is.

## Example

In this example, the following data is given, in a table named `employees`:

| employee_id | salary  |
| ----------- | ------- |
| 1           | 3500.27 |
| 2           | 1530.90 |
| 3           | 3025.10 |
| 4           | 5040.50 |

> The `FLOOR()` function can be used to calculate the largest integer that is less than or equal to a given value:

Reviewer: ```suggestion
The `FLOOR()` function can be used to return the largest integer that is less than or equal to a given value:
```

```sql
SELECT employee_id, FLOOR(salary) AS floored_salary
FROM employees;
```

> In this SQL statement, the `salary` values are floored for each employee to calculate the `floored_salary`. The [`AS`](https://www.codecademy.com/resources/docs/sql/commands/as) keyword is used to rename the `FLOOR(salary)` column to `floored_salary` in the output.

Reviewer: ```suggestion
In this SQL statement, the `salary` values are floored for each employee to get the `floored_salary`. The [`AS`](https://www.codecademy.com/resources/docs/sql/commands/as) keyword is used to name the resulting column `floored_salary` in the output.
```
